apps/api/src/statistics/statistics.service.ts

```diff
@@ -1,12 +1,162 @@
+
+import type { StatsByMonth } from "./schemas/userStats.schema";
 
 @Injectable()
 export class StatisticsService {
-  constructor(private statisticsRepository: StatisticsRepository) {}
+  constructor(
+    private statisticsRepository: StatisticsRepository,
+    private lessonsRepository: LessonsRepository,
+  ) {}
 
   async getUserStats(userId: string) {
-    return this.statisticsRepository.getUserStats(userId);
+    const coursesStatsByMonth: StatsByMonth[] =
+      await this.statisticsRepository.getCoursesStatsByMonth(userId);
+    const coursesTotalStats = this.calculateTotalStats(coursesStatsByMonth);
+
+    const lessonsStatsByMonth: StatsByMonth[] =
+      await this.statisticsRepository.getLessonsStatsByMonth(userId);
+    const lessonsTotalStats = this.calculateTotalStats(lessonsStatsByMonth);
+
+    const quizStats = await this.statisticsRepository.getQuizStats(userId);
+
+    // TODO: decide what to do when there is no last lesson
+    const lastLessonItem = await this.lessonsRepository.getLastLessonItemForUser(userId);
+    const [lastLesson] = await this.lessonsRepository.getLessonsDetails(
+      userId,
+      lastLessonItem.courseId,
+      lastLessonItem.lessonId,
+    );
+    const lastLessonDetails = await this.lessonsRepository.getLessonForUser(
+      lastLessonItem.courseId,
+      lastLessonItem.lessonId,
+      userId,
+    );
+
+    const activityStats = await this.statisticsRepository.getActivityStats(userId);
+
+    return {
+      courses: this.formatStats(coursesStatsByMonth),
+      lessons: this.formatStats(lessonsStatsByMonth),
+      quizzes: {
+        totalAttempts: quizStats.totalAttempts,
+        totalCorrectAnswers: quizStats.totalCorrectAnswers,
+        totalWrongAnswers: quizStats.totalWrongAnswers,
+        totalQuestions: quizStats.totalQuestions,
+        averageScore: quizStats.averageScore,
+        uniqueQuizzesTaken: quizStats.uniqueQuizzesTaken,
+      },
+      averageStats: {
+        lessonStats: lessonsTotalStats,
+        courseStats: coursesTotalStats,
+      },
+      lastLesson: {
+        ...lastLesson,
+        enrolled: lastLessonDetails.enrolled,
+      },
+      streak: {
+        current: activityStats.currentStreak ?? 0,
+        longest: activityStats.longestStreak ?? 0,
+        activityHistory: activityStats?.activityHistory || {},
+      },
+    };
+  }
+
+  async updateUserActivity(userId: string) {
+    const today = startOfDay(new Date());
+    const formatedTodayDate = format(today, "yyyy-MM-dd");
+
+    const currentStats = await this.statisticsRepository.getActivityStats(userId);
+
+    const lastActivityDate = currentStats?.lastActivityDate
+      ? startOfDay(new Date(currentStats.lastActivityDate))
+      : null;
+
+    const newCurrentStreak = (() => {
+      if (!lastActivityDate) return 1;
+
+      const daysDiff = differenceInDays(today, lastActivityDate);
+
+      if (daysDiff === 0) return currentStats?.currentStreak ?? 1;
```

Comment:
Maybe we should use switch case here, it will be clearer and will work better if there are more conditions in the future

`switch (daysDiff) {
  case 0:
    return currentStats?.currentStreak ?? 1;
  case 1:
    return (currentStats?.currentStreak ?? 0) + 1;
  default:
    return 1;
}
`